# Bonus: Temperature Analysis I

In [1]:
# Ignore SQLITE warnings related to Decimal numbers in the hawaii database
import warnings
warnings.filterwarnings('ignore')

In [2]:
# Import dependencies
from scipy import stats

In [3]:
import numpy as np
import pandas as pd
import datetime as dt

In [4]:
# Python SQL toolkit and Object Relational Mapper
import sqlalchemy
from sqlalchemy.ext.automap import automap_base
from sqlalchemy.orm import Session
from sqlalchemy import create_engine, func

In [5]:
# Using either SQLAlchemy or pandas, get the requested data per the README
# Store filepath in a variable
hawaii_file = "Resources/hawaii_measurements.csv"

In [6]:
# Read our Data file with the pandas library
# Not every CSV requires an encoding, but be aware this can come up
hawaii_file_df = pd.read_csv(hawaii_file, encoding="ISO-8859-1")
hawaii_file_df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [7]:
# Putting the month from the date column in another column to just look for June and December tobs mean
#pandas datetimeindex 
hawaii_file_df['month'] = pd.DatetimeIndex(hawaii_file_df['date']).month
hawaii_file_df.head()

station        date  prcp  tobs  month
0  USC00519397  2010-01-01  0.08    65      1
1  USC00519397  2010-01-02  0.00    63      1
2  USC00519397  2010-01-03  0.00    74      1
3  USC00519397  2010-01-04  0.00    76      1
4  USC00519397  2010-01-06   NaN    73      1

In [8]:
hawaii_file_df.dtypes

station     object
date        object
prcp       float64
tobs         int64
month        int64
dtype: object

In [9]:
# hawaii_file_df.dropna(inplace=True)
hawaii_file_df.fillna(value=' ', inplace=True)
hawaii_file_df

station        date  prcp  tobs  month
0      USC00519397  2010-01-01  0.08    65      1
1      USC00519397  2010-01-02     0    63      1
2      USC00519397  2010-01-03     0    74      1
3      USC00519397  2010-01-04     0    76      1
4      USC00519397  2010-01-06          73      1
...            ...         ...   ...   ...    ...
19545  USC00516128  2017-08-19  0.09    71      8
19546  USC00516128  2017-08-20          78      8
19547  USC00516128  2017-08-21  0.56    76      8
19548  USC00516128  2017-08-22   0.5    76      8
19549  USC00516128  2017-08-23  0.45    76      8

[19550 rows x 5 columns]

In [10]:
june_tobs = hawaii_file_df.loc[hawaii_file_df['month'] == 6]['tobs'].mean()
june_tobs

74.94411764705882

In [11]:
dec_tobs = hawaii_file_df.loc[hawaii_file_df['month'] == 12]['tobs'].mean()
dec_tobs

71.04152933421226

In [14]:
# Use t-test to determine if difference in means is statistically significant
a = hawaii_file_df.query('month == "6"')['tobs']
b = hawaii_file_df.query('month == "12"')['tobs']
stats.ttest_ind(a, b, equal_var=False)

Ttest_indResult(statistic=31.355036920962423, pvalue=4.193529835915755e-187)

## Discuss which t-test to use and why and what your results mean.
## Use the t-test to determine whether the difference in the means, if any,
## is statistically significant. Will you use a paired t-test, or an unpaired t-test? Why?

In this case, both of the averages(means) of the temperatures in June and December 
are very close in value(June = 74.94 and December = 71.04), and they are two independent groups
in which the best t-test to apply is the Independent T-test. And the result of the T-test has a 
p-value of 4.193529835915755e-187 which is a very low p-value in statistics, and it represents 
a good significant relation and representation between the data.